api/include/opentelemetry/std/variant.h

```diff
@@ -0,0 +1,232 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "opentelemetry/version.h"
+
+#include <cstddef>
+#include <memory>
+#include <utility>
+#include <variant>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+// Standard Type aliases in nostd namespace
+namespace nostd
+{
+
+// nostd::variant<...>
+template <class... _Types>
+using variant = std::variant<_Types...>;
+
+#if defined(__APPLE__) && defined(_LIBCPP_USE_AVAILABILITY_APPLE)
+// Apple Platforms provide std::bad_variant_access only in newer versions of OS.
```

Comment:
Additional complexity here is that `std::variant` in Apple LLVM is only available with certain version of Mac OS X, and the standard version detection (C++17 and above) - won't work.. We have to stub exception handling for Mac, in order to benefit from the library implementation + methods for parts that are missing.

Reply:
I added a separate note on this in the markdown doc.